src/shared/components/table/state/tableReducer.ts

```diff
@@ -0,0 +1,220 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { ReactNode } from 'react';
+
+export enum SortingDirection {
+  ASC = 'ascending',
+  DESC = 'descending',
+  DEFAULT = 'default' // Do we need this?
+}
+
+export type SortedColumn = {
+  columnId: string;
+  sortedDirection: SortingDirection;
+};
+
+export enum TableAction {
+  DEFAULT = 'default',
+  FIND_IN_TABLE = 'find_in_table',
+  FILTERS = 'filters',
+  SHOW_HIDE_COLUMNS = 'show_hide_columns',
+  SHOW_HIDE_ROWS = 'show_hide_rows',
+  DOWNLOAD_SHOWN_DATA = 'download_shown_data',
+  DOWNLOAD_ALL_DATA = 'download_all_data',
+  RESTORE_DEFAULTS = 'restore_defaults'
+}
+
+export type TableCellData = ReactNode;
+
+export type TableRowData = {
+  cells: TableCellData[];
+  expandedContent?: ReactNode;
+};
+
+export type TableData = TableRowData[];
+
+export type TableCellRendererParams = {
+  rowId: string;
+  rowData: TableRowData;
+  cellData: TableCellData;
+};
+
+export type IndividualColumn = {
```

Comment:
Do you think it would be good to move all the table types into a file?

Reply:
That's done